daemon/start.go

```diff
@@ -158,6 +158,12 @@ func (daemon *Daemon) containerStart(container *container.Container) (err error)
 			container.SetExitCode(126)
 		}
 
+		// attempted to mount a file onto a directory, or a directory onto a file, maybe from user specified bind mounts
+		if strings.Contains(errDesc, syscall.ENOTDIR.Error()) {
+			errDesc += ": check that all specified host paths exist and are the correct file type for all volumes"
```

Comment:
I like that. I started out this way (kind of) but changed it since the error itself could come from anything (not just user specified mounts)... but I think as you have it works well.